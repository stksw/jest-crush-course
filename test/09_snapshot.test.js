/**
 * @jest-environment jest-environment-jsdom
 */
import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import Button from '../09_Button';

describe('snapshot test1', () => {
  it('renders correctly with react-test-renderer', () => {
    const button = renderer.create(<Button />);
    const tree = button.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('snapshot test2', () => {
  test('renders correctly with @testing-library/react', () => {
    const button = render(<Button />);
    expect(button.container).toMatchSnapshot();

    fireEvent.click(button.getByText('ON'));
    expect(button.container).toMatchSnapshot();

    fireEvent.click(button.getByText('OFF'));
    expect(button.container).toMatchSnapshot();
  });
});
