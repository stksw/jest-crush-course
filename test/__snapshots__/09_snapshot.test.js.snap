// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test renders correctly with @testing-library/react 1`] = `undefined`;

exports[`snapshot test renders correctly with react-test-renderer 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "green",
      "color": "white",
    }
  }
>
  ON
</button>
`;

exports[`snapshot test1 renders correctly with react-test-renderer 1`] = `
<button
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "green",
      "color": "white",
    }
  }
>
  ON
</button>
`;

exports[`snapshot test2 renders correctly with @testing-library/react 1`] = `
<div>
  <button
    style="background-color: green; color: white;"
  >
    ON
  </button>
</div>
`;

exports[`snapshot test2 renders correctly with @testing-library/react 2`] = `
<div>
  <button
    style="background-color: green; color: white;"
  >
    OFF
  </button>
</div>
`;

exports[`snapshot test2 renders correctly with @testing-library/react 3`] = `
<div>
  <button
    style="background-color: green; color: white;"
  >
    ON
  </button>
</div>
`;
